src/public/services/FormSgSdkService.ts

```diff
@@ -0,0 +1,16 @@
+import formSgSdk from '@opengovsg/formsg-sdk'
+import { PackageMode } from '@opengovsg/formsg-sdk/dist/types'
+
+import { TRANSACTION_EXPIRE_AFTER_SECONDS } from '../../shared/util/verification'
+
+// Reading from process.env.FORMSG_SDK_MODE defaults to undefined
+// This is because browsers are sandboxed and hence have to manually check on window...
+const sdkMode = (((window as unknown) as { formsgSdkMode: PackageMode })
```

Comment:
window is known as a global - the problem comes because window is interpreted as having `typeof Window`, which is a problem because this property is set elsewhere, so we need to typecast for ts to accept the property access

Reply:
if that's the case then we should think of a better approach to accessing the window variables. we shouldn't keep typecasting all over the place when the frontend is in TS. would it be possible to create a separate file which checks for the presence of the env vars in `window` and exports them? perhaps we can use a util similar to the existing `hasProp` to ensure that the window variables can be accessed in a type-safe manner?

Reply:
i do agree with accessing the window variables but i think `hasProp` is the wrong way to go around it. Because of injection, the type of window should be known at the start and hence, i think it makes more sense to define a custom type and export (!!!) window as the custom type so that
 
1. all property access is safe 
2. no need to keep checking for a specific property 

wdyt about this approach?

Reply:
so I did consider your approach, but I thought exporting just the env vars instead of the entire `window` object would make more sense.

let's say we export a custom window type called `FormsgWindow`, and in some future React component, I access `window.localStorage`. would you request me to change it to `FormsgWindow.localStorage`, even though the env vars are not even relevant to this part of the code? personally if I was the reviewer, I wouldn't ask for this change for the sake of it.

this means we would only use `FormsgWindow` when needing to access the env vars...but that means exporting the `window` methods is useless, and we might as well just export the env vars. wydt?

Reply:
i think we are on the same tangent but your interpretation is different! because here, we are accessing _specific_ 
 and _custom_ properties that are set on `window`, we **need** the custom export to avoid ts rightfully kicking up a fuss. 

however, if we were to access _inherent_ properties/methods of the `window` object,  this wouldn't be an issue (and wouldn't even require the import) - the export of `window` here is solely to ensure the caller (and ts) that yes, we've done the checking and the property does indeed exist on window

Reply:
but why export an entire custom window object instead of just exporting the env vars?

Reply:
oh i think i get what you're saying here. rather than export the entire window object, we can export the injected env vars as an object and then access the required env vars from this object. this is what you're proposing right, not that we export the injected values as a type

Reply:
yep